lib/puppet/transaction/persistence.rb

```diff
@@ -56,18 +56,21 @@ def load
       begin
         result = Puppet::Util::Yaml.load_file(filename)
       rescue Puppet::Util::Yaml::YamlLoadError => detail
-        Puppet.err "Transaction store file #{filename} is corrupt (#{detail}); replacing"
+        Puppet.log_exception(detail, "Transaction store file #{filename} is corrupt (#{detail}); replacing")
```

Comment:
Yes, and in an oh-shit moment, I found a bug in property.rb with how many args you're passing. I'll fix this case here, and fix the other bug too.